pkg/cmd/server/origin/handlers.go

```diff
@@ -168,6 +176,98 @@ func namespacingFilter(handler http.Handler, contextMapper kapi.RequestContextMa
 	})
 }
 
+// versionSkewFilter adds a filter that may deny PUT an POST requests since they stomp entire resource objects from skewed
+// oc clients, since we know that those clients will strip unknown fields which can lead to unexpected outcomes
+func (c *MasterConfig) versionSkewFilter(binaryInfo version.Info, handler http.Handler) http.Handler {
+	skewedClientPolicy := c.Options.PolicyConfig.SkewedClientPolicy
+	if skewedClientPolicy == configapi.AllowAll {
+		return handler
+	}
+	seg := strings.SplitN(binaryInfo.GitVersion, "-", 2)
+	serverVersionNumber := seg[0][1:]
+	serverCommit := binaryInfo.GitCommit
+	if len(serverCommit) > 7 {
+		serverCommit = serverCommit[:7]
+	}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		if req.Method != "PUT" && req.Method != "POST" {
```

Comment:
Not patch.  Patches are non-encoded json that avoids any field stripping logic.  That stripping logic is the only reason to block skewed usage.